File: archived/DisablingScreenCapture/js/js/scenario2-disable.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var page = WinJS.UI.Pages.define("/html/scenario2-disable.html", {
        ready: function (element, options) {
            // Disable screen capture when this page is loaded.
            Windows.UI.ViewManagement.ApplicationView.getForCurrentView().isScreenCaptureEnabled = false;
        },
        unload: function () {
            // Re-enable screen capture when this page is unloaded.
            Windows.UI.ViewManagement.ApplicationView.getForCurrentView().isScreenCaptureEnabled = true;
        }
    });
})();

File: Samples/DisablingScreenCapture/cppwinrt/Scenario2_Disable.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario2_Disable.h"
#include "Scenario2_Disable.g.cpp"

using namespace winrt;
using namespace winrt::Windows::Foundation;
using namespace winrt::Windows::UI::Xaml;
using namespace winrt::Windows::UI::Xaml::Navigation;

namespace winrt::SDKTemplate::implementation
{
    Scenario2_Disable::Scenario2_Disable()
    {
        InitializeComponent();
    }

    void Scenario2_Disable::OnNavigatedTo(NavigationEventArgs const&)
    {
        // Disable screen capture when the user navigates to this page.  
        Windows::UI::ViewManagement::ApplicationView::GetForCurrentView().IsScreenCaptureEnabled(false);
    }

    void Scenario2_Disable::OnNavigatedFrom(NavigationEventArgs const&)
    {
        // Re-enable screen capture when the user navigates away from this page.
        Windows::UI::ViewManagement::ApplicationView::GetForCurrentView().IsScreenCaptureEnabled(true);
    }
}
